src/validr/_validator.pyx

```diff
@@ -115,6 +115,53 @@ def validator(bint string=False):
     return decorator
 
 
+_UNIQUE_CHECK_ERROR_MESSAGE = """
+Unable to check unique! only list of types below are available:
+    1. scalar type, eg: bool, int, float, str
+    2. list of scalar type
+    3. dict of scalar type value
+There are no general and fast unique check for non-hashable types,
+you should do unique check by yourself in this case.
+""".strip()
+
+
+def _is_scalar(validator):
+    return validator != 'dict' and validator != 'list'
+
+
+def _key_func_of_schema(schema):
+    if schema is None:
+        raise SchemaError(_UNIQUE_CHECK_ERROR_MESSAGE)
+
+    if schema.validator == 'dict':
+        if schema.items is None:
+            raise SchemaError(_UNIQUE_CHECK_ERROR_MESSAGE)
+        keys = []
+        for k, v in schema.items.items():
+            if not _is_scalar(v.validator):
```

Comment:
What if item is a list of scalar types? 
I think in the case check should sink recursively. So, if the 'leaf' is hashable, then we check it's 'branch' and so on.

Reply:
List or dict, of cause

Reply:
Yes, recursively is better.